core/trino-main/src/main/java/io/trino/execution/buffer/PagesSerdeFactory.java

```diff
@@ -27,20 +38,61 @@ public class PagesSerdeFactory
 
     private final BlockEncodingSerde blockEncodingSerde;
     private final boolean compressionEnabled;
+    private final CompressionCodec compressionCodec;
 
     public PagesSerdeFactory(BlockEncodingSerde blockEncodingSerde, boolean compressionEnabled)
+    {
+        this(blockEncodingSerde, compressionEnabled, LZ4);
+    }
+
+    public PagesSerdeFactory(BlockEncodingSerde blockEncodingSerde, boolean compressionEnabled, CompressionCodec compressionCodec)
     {
         this.blockEncodingSerde = requireNonNull(blockEncodingSerde, "blockEncodingSerde is null");
         this.compressionEnabled = compressionEnabled;
+        this.compressionCodec = requireNonNull(compressionCodec, "compressionCodec is null");
     }
 
     public PageSerializer createSerializer(Optional<SecretKey> encryptionKey)
     {
-        return new PageSerializer(blockEncodingSerde, compressionEnabled, encryptionKey, SERIALIZED_PAGE_DEFAULT_BLOCK_SIZE_IN_BYTES);
+        return new PageSerializer(blockEncodingSerde, compressionEnabled, createCompressor(compressionCodec), encryptionKey, SERIALIZED_PAGE_DEFAULT_BLOCK_SIZE_IN_BYTES);
     }
 
     public PageDeserializer createDeserializer(Optional<SecretKey> encryptionKey)
     {
-        return new PageDeserializer(blockEncodingSerde, compressionEnabled, encryptionKey, SERIALIZED_PAGE_DEFAULT_BLOCK_SIZE_IN_BYTES);
+        return new PageDeserializer(blockEncodingSerde, compressionEnabled, createDecompressor(compressionCodec), encryptionKey, SERIALIZED_PAGE_DEFAULT_BLOCK_SIZE_IN_BYTES);
+    }
+
+    public static Compressor createCompressor(CompressionCodec compressionCodec)
+    {
+        switch (compressionCodec) {
+            case NONE:
+                return null;
```

Comment:
I don't like the fact that we are returning null here.

What's the sense of the combination `compressionEnabled == true && compressionCodec == NONE` ?

Reply:
Let's drop NONE and use the `compressionEnabled` information that states explicitly that compression is not used

Reply:
something along these lines: https://github.com/wendigo/trino/commit/c2a7f64b0e870224921a1f45d24eb6fd4bdee93d#diff-6e2a6eb5c7e2eba83c56db26e3868fcc61a0263a479bc052dd67d33ba864d3afR65

Reply:
> something along these lines: [wendigo@c2a7f64#diff-6e2a6eb5c7e2eba83c56db26e3868fcc61a0263a479bc052dd67d33ba864d3afR65](https://github.com/wendigo/trino/commit/c2a7f64b0e870224921a1f45d24eb6fd4bdee93d#diff-6e2a6eb5c7e2eba83c56db26e3868fcc61a0263a479bc052dd67d33ba864d3afR65)

Thank you very much for the codes~ yes the NONE enum is useless in this case, I just copied it from `io.trino.orc.metadata.CompressionKind` at the beginning~ I have deleted it this time